core/domain/story_domain.py

```diff
@@ -369,6 +374,10 @@ def validate(self):
         if self.thumbnail_filename and self.thumbnail_bg_color is None:
             raise utils.ValidationError(
                 'Chapter thumbnail background color is not specified.')
+        if self.thumbnail_filename is not None and (
+                self.thumbnail_size_in_bytes == 0):
+            raise utils.ValidationError(
```

Comment:
@anandwana001 
Add tests in story_domain_test.py for this condition

Reply:
done